packages/react/src/auth/useSession.ts

```diff
@@ -69,6 +75,7 @@ export function useSession<
 
     const opts: any = {
       suspense: true,
```

Comment:
does this have anything to do with this suspense?

Reply:
ok so after some reading I think the issue is that we have a `<Suspense>` boundary at the top level of the app and what happens on hydration is the `useSession` hook suspends which suspends up to the top level boundary; if before that suspense resolves some other state changes then we will get this hydration error

I think maybe the better solution is to move the suspense boundary closer to where the `useSession` is called